src/Model/Profile.php

```diff
@@ -166,7 +166,17 @@ public static function load(App $a, $nickname, array $profiledata = [], $show_co
 			}
 		}
 
-		$profile = !empty($user['uid']) ? User::getOwnerDataById($user['uid'], false) : [];
+		if (empty($user['uid'])) {
+			$profile = [];
+		} else {
+			$contact_id = Contact::getIdForURL(Strings::normaliseLink(DI::baseurl() . '/profile/' . $nickname), local_user());
+			$profile = array_merge(
+				$user,
+				Contact::getById($contact_id),
+				Profile::getByUID($user['uid']),
+			);
```

Comment:
What is this code meant to do - in difference to calling `User::getOwnerDataById`? If you just want to fetch the contact id for the user that you are just visiting, then you could use `$profile['url']` from the `User::getOwnerDataById` call.

Reply:
Initially I _did_ write it so that all I did was set the correct contact ID for 'cid' and change the 'self' value, but I found myself writing a huge comment to explain what was going on. And it bothered me that the rest of the contact data would be coming from the 'self' contact, which sort of meant betting on nothing downstream expecting a "correct" contact entry in $profile. So I decided to try to emulate a version of the "owner-view" record that would be returned from getOwnerDataById, but with the correct contact entry for the logged in user instead of the 'self' entry.

Reply:
Reviewing the sidebar function, I've noticed this time that it goes out to retrieve the contact entry from the database based on 'cid', so I might actually be able to remove the Contact::getById call there. I've just done so on my instance and nothing seems to have broken yet.

Reply:
I'm not a huge fan of arbitrarily creating the profile array, since it is used at several places and some functions possibly rely upon fields that aren't created in this case. So I really would appreciate a solution with a small impact, preferably directly in the `sidebar` function.

Reply:
Hmm. While nothing has apparently broken on my instance, I also see that `sidebar` wants to be able to examine a "network" key on $profile, which comes from the `contact` table. So probably better to leave that line in and stay as close to the original `owner-view` row as possible.

Reply:
Alright then. I'll issue a pull request to roll back my changes and try to limit my edits to the sidebar function.

Reply:
That part of the code is really ancient stuff. And it hadn't gone better over the years - and I'm sure that a big chunk of ugliness had been produced by my very own hands :-)